stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
-    )  #: The end longitude (:class:`float`)
-    climb_category = Attribute(
-        int, (SUMMARY, DETAILED)
-    )  # 0-5, lower is harder
-    city = Attribute(str, (SUMMARY, DETAILED))  #: The city this segment is in.
-    state = Attribute(
-        str, (SUMMARY, DETAILED)
-    )  #: The state this segment is in.
-    country = Attribute(
-        str, (SUMMARY, DETAILED)
-    )  #: The country this segment is in.
-    private = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  #: Whether this is a private segment.
-    starred = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  #: Whether this segment is starred by authenticated athlete
+    _latlng_check = validator('start_latlng', 'end_latlng', allow_reuse=True, pre=True)(check_valid_location)
 
-    athlete_segment_stats = EntityAttribute(
-        AthleteSegmentStats, (DETAILED,)
-    )  #: Undocumented attrib holding stats for current athlete.
+    _latlng_extensions = extend_types('start_latlng', 'end_latlng', model_class=LatLon)
+    _map_extensions = extend_types('map', model_class=Map)
+    _segment_stat_extension = extend_types('athlete_segment_stats', model_class=AthleteSegmentStats)
+    _pr_effort_extension = extend_types('athlete_pr_effort', model_class=AthletePrEffort)
 
-    # detailed attribs
-    created_at = TimestampAttribute(
-        (DETAILED,)
-    )  #: :class:`datetime.datetime` when was segment created.
-    updated_at = TimestampAttribute(
-        (DETAILED,)
-    )  #: :class:`datetime.datetime` when was segment last updated.
-    total_elevation_gain = Attribute(
-        float, (DETAILED,), units=uh.meters
-    )  #: What is total elevation gain for segment.
-    map = EntityAttribute(
-        Map, (DETAILED,)
-    )  #: :class:`stravalib.model.Map` object for segment.
-    effort_count = Attribute(
-        int, (DETAILED,)
-    )  #: How many times has this segment been ridden.
-    athlete_count = Attribute(
-        int, (DETAILED,)
-    )  #: How many athletes have ridden this segment
-    hazardous = Attribute(
-        bool, (DETAILED,)
-    )  #: Whether this segment has been flagged as hazardous
-    star_count = Attribute(
-        int, (DETAILED,)
-    )  #: number of stars on this segment.
-    pr_time = Attribute(int, (DETAILED,))  #: pr time for athlete
-    starred_date = TimestampAttribute((DETAILED,))  #: datetime when be starred
-    athlete_pr_effort = EntityAttribute(AthletePrEffort, (DETAILED,))
-    elevation_profile = Attribute(str, (SUMMARY, DETAILED))
 
-    @property
-    def leaderboard(self):
-        """
-        The :class:`stravalib.model.SegmentLeaderboard` object for this segment.
-        """
-        if self._leaderboard is None:
-            self.assert_bind_client()
-            if self.id is not None:
-                self._leaderboard = self.bind_client.get_segment_leaderboard(
-                    self.id
-                )
-        return self._leaderboard
+class SegmentEffortAchievement(BaseModel):
```

Comment:
This is a small example of how to add undocumented entities/fields to our model. Following PRs should add the undocumented fields from the legacy model.